modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -377,7 +379,22 @@ public String toModelFilename(String name) {
 
     @Override
     public String toModelImport(String name) {
-        return modelPackage() + "/" + toModelFilename(name);
+        // with modelNamePrefix/modelNameSuffix, name already includes
+        // the prefix/suffix.
+        // for example if modelNamePrefix is 'Swg' and the model name 'Pet',
+        // name is already SwgPet. toModelFilename(name) is SwgSwgPet instead
+        // of SwgPet.
+        String filename = toModelFilename(name);
+
+        if (!StringUtils.isEmpty(modelNamePrefix)) {
+            filename = StringUtils.removeStart(filename, modelNamePrefix);
+        }
+
+        if (!StringUtils.isEmpty(modelNameSuffix)) {
+            filename = StringUtils.removeEnd(filename, modelNameSuffix);
+        }
+
+        return modelPackage() + "/" + filename;
```

Comment:
Filename is capitalised. `import { SwgError} from '../model/SwgError';` leads to issues on case sensitive OS at least.